src/pages/IDO/index.tsx

```diff
@@ -0,0 +1,172 @@
+const PngIcon = styled.div`
+  transition: transform 0.3s ease;
+  :hover {
+    transform: rotate(-5deg);
+  }
+`
+const activeClassName = 'ACTIVE'
+
+const StyledExternalLink = styled(ExternalLink).attrs({
+  activeClassName
+})<{ isActive?: boolean }>`
+  ${({ theme }) => theme.flexRowNoWrap}
+  align-items: left;
+  border-radius: 3rem;
+  outline: none;
+  cursor: pointer;
+  text-decoration: none;
+  color: ${({ theme }) => theme.text2};
+  font-size: 1rem;
+  width: fit-content;
+  margin: 0 12px;
+  font-weight: 500;
+
+  &.${activeClassName} {
+    border-radius: 12px;
+    font-weight: 600;
+    color: ${({ theme }) => theme.text1};
+  }
+
+  :hover,
+  :focus {
+    text-decoration: none;
+    color: ${({ theme }) => darken(0.1, theme.text1)};
+  }
+
+  ${({ theme }) => theme.mediaWidth.upToExtraSmall`
+      display: none;
+`}
+`
+const IDOs = styled.div`
+  padding: 0.75rem 1rem;
+  width: 100%;
+  margin-top: 1rem;
+  border-radius: 12px;
+  display: grid;
+  grid-template-columns: 60px 1fr 100px;
+  align-items: center;
+  text-align: left;
+  outline: none;
+  cursor: pointer;
+  color: ${({ theme }) => theme.text1};
+  text-decoration: none;
+  background-color: ${({ theme }) => theme.bg1};
+  &:focus {
+    background-color: ${({ theme }) => darken(0.05, theme.bg1)};
+  }
+  &:hover {
+    background-color: ${({ theme }) => darken(0.05, theme.bg1)};
+  }
+`
+
+export default function IDO() {
+  const { t } = useTranslation()
+
+  return (
+    <PageWrapper gap="lg" justify="center">
+      {/* This is the top section of the page */} 
+      <TopSection gap="md">
+        <VoteCard>
+          <CardBGImage />
+          <CardNoise />
+          <CardSection>
+            <AutoColumn gap="md">
+              <RowBetween>
+                <TYPE.white fontWeight={600}>{t('IDOPage.pangolinIDOs')}</TYPE.white>
+              </RowBetween>
+              <RowBetween>
+                <TYPE.white fontSize={14}>{t('IDOPage.pangolinIDODescription')}</TYPE.white>
+              </RowBetween>
+              <RowBetween>
+              </RowBetween>
+            </AutoColumn>
+          </CardSection>
+          <CardBGImage />
+          <CardNoise />
+        </VoteCard>
+      </TopSection>
+      <TopSection gap="2px">
+        <WrapSmall> {/* This is the title section of the IDO list */}
+          <TYPE.mediumHeader style={{ margin: '0.5rem 0.5rem 0.5rem 0', flexShrink: 0 }}>
+            {t('IDOPage.upcomingIDOs')}
+          </TYPE.mediumHeader> 
+        </WrapSmall>
+
+        {/* We can use this when there are no upcoming IDOs 
+        {allProposals?.length === 0 && (
+          <EmptyProposals>
+            <TYPE.body style={{ marginBottom: '8px' }}>{t('IDOPage.noIDOsFound')}</TYPE.body>
+            <TYPE.subHeader>
+              <i>{t('IDOPage.IDOAnnounced')}</i>
+            </TYPE.subHeader>
+          </EmptyProposals>
+        )} 
+        
+        */}
+        {
+          <IDOs>
```

Comment:
These should be pulled from a config list so that adding / removing IDOs is easy in the future. Especially if somebody non technical is managing this. We would iterate the config object here and it would contain properties like:

title, image link, description, twitter link, ido date, etc.

Reply:
placed an array in constants/index.ts where items can be added or change their status . If there are no upcoming IDOs, it shows a message. I also moved the images from the src directory to the public/images directory to be able to access them in the component.

Reply:
Excellent! I'd recommend moving the `IDO_LIST` into its own file under constants/ like constants/idos.ts to avoid future merge conflicts. that index.ts file gets updated a LOT by new rewards. also to add a constant in there for "Upcoming" and "Ended" or so to easily reference this in the display logic

Reply:
Moved the IDO_LIST into its own file. Added an interface for the array items in the list. Added the IDO_STATUS_UPCOMMING and IDO_STATUS_ENDED constants

Also rearranged the IDO list to place Kalao on top with an updated announcement url